lib/src/services/app_coin_services.dart

```diff
@@ -0,0 +1,31 @@
+import 'dart:convert';
+
+import 'package:http/http.dart' as http;
+
+import '../constants/strings.dart';
+import '../models/coins.dart';
+import 'coin_service.dart';
+
+class AppCoinService extends CoinService {
+  AppCoinService(http.Client client) : super(client);
+
+  Future<List<Coins>> getCoinsFromAPI(
+      {required String currency,
+      required int start,
+      required int limit,
+      required bool sparkline}) async {
+    var url = Uri.parse(StringData.FINAL_API_URL +
+        '/markets?vs_currency=$currency&order=market_cap_desc&per_page=$limit&page=$start&sparkline=$sparkline');
```

Comment:
for now, this is ok, but it will be better if you use some config method to hide this API information.
you can take a look at Mr.Hai example or your classmate repos

Reply:
I fixed